src/main/java/seedu/fridgefriend/command/ListCommand.java

```diff
@@ -23,50 +23,57 @@ public ListCommand(String description) {
     }
 
     @Override
-    public void execute(List<Food> fridge) {
+    public void execute() throws InvalidInputException {
         if (category.equals("")) {
-            listEverythingInFridge(fridge);
+            listEverythingInFridge();
         } else {
-            listByCategory(fridge);
+            listByCategory();
         }
     }
 
-    private void listByCategory(List<Food> fridge) {
-        try {
-            checkIsValidCategory();
-            System.out.println("These are the " + foodCategory + " in your fridge:");
-            for (int i = 0; i < fridge.size(); i++) {
-                matchCategory(i, fridge);
-            }
-            System.out.println();
-        } catch (InvalidInputException invalidInputException) {
-            System.out.println("Sorry my friend, please enter a valid food category.\n");
+    private void listByCategory() throws InvalidInputException {
+        checkIsValidCategory();
+        String message = "These are the " + foodCategory + " in your fridge:\n";
+        for (int i = 0; i < Fridge.getSize(); i++) {
+            message += matchCategory(i);
```

Comment:
Would it be better to SLAP this method (i.e. make them all methods?)

Reply:
Will refactor this method